client/my-sites/sharing/main.jsx

```diff
@@ -1,90 +1,95 @@
-			filters = [];
+class Sharing extends Component {
+	getFilters() {
+		const { canManageOptions, site, translate } = this.props;
+		const pathSuffix = site ? '/' + site.slug : '';
+		const filters = [];
 
 		// Include Connections link if all sites are selected. Otherwise,
 		// verify that the required Jetpack module is active
 		if ( ! site || ! site.jetpack || site.isModuleActive( 'publicize' ) ) {
-			filters.push( { title: this.translate( 'Connections' ), path: '/sharing' + pathSuffix, id: 'sharing-connections' } );
+			filters.push( {
+				id: 'sharing-connections',
+				path: '/sharing' + pathSuffix,
+				title: translate( 'Connections' ),
+			} );
 		}
 
 		// Include Sharing Buttons link if a site is selected and the
 		// required Jetpack module is active
-		if ( site && utils.userCan( 'manage_options', site ) &&
+		if ( site && canManageOptions &&
 			( ! site.jetpack ||
 				( site.isModuleActive( 'sharedaddy' ) && site.versionCompare( '3.4-dev', '>=' ) )
 			)
 		) {
-			filters.push( { title: this.translate( 'Sharing Buttons' ), path: '/sharing/buttons' + pathSuffix, id: 'sharing-buttons' } );
+			filters.push( {
+				id: 'sharing-buttons',
+				path: '/sharing/buttons' + pathSuffix,
+				title: translate( 'Sharing Buttons' ),
+			} );
 		}
 
 		return filters;
-	},
+	}
+
+	render() {
+		const filters = this.getFilters();
+		const selected = find( filters, { path: this.props.path } );
 
-	render: function() {
 		return (
 			<Main className="sharing">
+				<DocumentHead title={ this.props.translate( 'Sharing', { textOnly: true } ) } />
 				<SidebarNavigation />
-				<SectionNav selectedText={ this.getSelectedText() }>
+				<SectionNav selectedText={ get( selected, 'title', '' ) }>
 					<NavTabs>
-						{ this.getFilters().map( function( filterItem ) {
-							return (
-								<NavItem
-									key={ filterItem.id }
-									path={ filterItem.path }
-									selected={ filterItem.path === this.props.path }
-								>
-									{ filterItem.title }
-								</NavItem>
-							);
-						}, this ) }
+						{ filters.map( ( { id, path, title } ) => (
+							<NavItem key={ id } path={ path } selected={ path === this.props.path }>
+								{ title }
+							</NavItem>
+						) ) }
 					</NavTabs>
 				</SectionNav>
 				<UpgradeNudge
-					title={ this.translate( 'No Ads with WordPress.com Premium' ) }
-					message={ this.translate( 'Prevent ads from showing on your site.' ) }
+					title={ this.props.translate( 'No Ads with WordPress.com Premium' ) }
+					message={ this.props.translate( 'Prevent ads from showing on your site.' ) }
 					feature="no-adverts"
 					event="sharing_no_ads"
 				/>
 				{ this.props.contentComponent }
 			</Main>
 		);
 	}
-} );
+}
+
+Sharing.propTypes = {
+	canManageOptions: PropTypes.bool,
+	contentComponent: PropTypes.node,
+	path: PropTypes.string,
+	site: PropTypes.object,
+	translate: PropTypes.func,
+};
+
+export default connect(
+	( state, { site } ) => ( {
```

Comment:
Maybe a little much to refactor in the context of this pull request, but it would be nice to drop the full site object in favor of the more specific selectors: `getSiteSlug`, `isJetpackSite`, `isJetpackModuleActive`, `isJetpackMinimumVersion`.